Libraries/Animated/src/AnimatedImplementation.js

```diff
@@ -1462,6 +1466,30 @@ function createAnimatedComponent(Component: any): any {
 
     componentDidMount() {
       this._propsAnimated.setNativeView(this.refs[refName]);
+
+      this.attachNativeEvents(this.props);
+    }
+
+    attachNativeEvents(nextProps) {
+      // TODO: Make sure we remove events properly.
+
+      // Make sure to get the scrollable node for components that implement
+      // `ScrollResponder.Mixin`.
+      const ref = this.refs[refName].getScrollableNode ?
+        this.refs[refName].getScrollableNode() :
+        this.refs[refName];
+
+      for (let key in nextProps) {
+        const prop = nextProps[key];
+        if (prop instanceof AnimatedEvent && prop.__isNative) {
+          // TODO: Map event names using the map in UIManager.
+          if (key === 'onScroll') {
```

Comment:
@brentvatne this approach will only work for "direct" events. That is because touch handling has a bubbling logic written in JS and we cannot determine which view is going to be responsible for handling the touch without running some JS code